source/Halibut/Util/TaskExtensionMethods.cs

```diff
@@ -24,5 +25,34 @@ public static void IgnoreUnobservedExceptions(this Task task)
                 },
                 TaskContinuationOptions.OnlyOnFaulted | TaskContinuationOptions.ExecuteSynchronously);
         }
+
+
+        public static IAsyncResult AsAsynchronousProgrammingModel<T>(
```

Comment:
This bit was stolen from [that Microsoft site](https://learn.microsoft.com/en-us/dotnet/standard/asynchronous-programming-patterns/interop-with-other-asynchronous-patterns-and-types)

Reply:
Could you note that in the code itself, so others will know what to read when they turn up to this?